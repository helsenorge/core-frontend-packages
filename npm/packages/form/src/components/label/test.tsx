import * as React from 'react';
import { mount } from 'enzyme';
import { Label } from '.';

describe('Gitt at Label skal vises', () => {
  describe('Når label skal mountes på siden', () => {
    it('Så vises den riktig med  default styling', () => {
      const wrapper = mount(<Label labelText={'my label'} />);
      expect(wrapper.render()).toMatchSnapshot();
    });
  });
  describe('Når label skal mountes på siden', () => {
    it('Så vises den riktig med custom className prop', () => {
      const wrapper = mount(<Label labelText={'my label'} className="test-css" />);
      expect(wrapper.render()).toMatchSnapshot();
    });
  });
  describe('Når label skal mountes på siden', () => {
    it('Så vises den riktig med custom className prop og sublabel', () => {
      const wrapper = mount(<Label labelText={'my label'} sublabelText={'my sublabel'} className="test-css" />);
      expect(wrapper.render()).toMatchSnapshot();
    });

    it('Så vises helpButton på siden', () => {
      const wrapper = mount(<Label labelText={'my label'} helpButton={<span className="help-button">{`help`}</span>} />);
      expect(wrapper.find('.help-button').length).toEqual(1);
    });
  });
});
