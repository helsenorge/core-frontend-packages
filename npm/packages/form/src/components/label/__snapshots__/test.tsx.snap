// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gitt at Label skal vises Når label skal mountes på siden Så vises den riktig med  default styling 1`] = `
<div
  class=""
>
  <label
    class="label"
  >
    my label
  </label>
</div>
`;

exports[`Gitt at Label skal vises Når label skal mountes på siden Så vises den riktig med custom className prop 1`] = `
<div
  class="test-css"
>
  <label
    class="label"
  >
    my label
  </label>
</div>
`;

exports[`Gitt at Label skal vises Når label skal mountes på siden Så vises den riktig med custom className prop og sublabel 1`] = `
<div
  class="test-css"
>
  <label
    class="label"
  >
    my label
    <span
      class="label__sublabel"
    >
      my sublabel
    </span>
  </label>
</div>
`;
